fs/grub/disk_io.c

```diff
@@ -125,36 +125,30 @@ int filepos;
 int filemax;
 
 //Cromwell hooks
-int
-rawread (int drive, int sector, int byte_offset, int byte_len, char *buf)
+int rawread(int drive, int sector, int byte_offset, int byte_len, char *buf)
```

Comment:
The original code also seems to do sanity checking.

See https://stuff.mit.edu/afs/sipb/project/hurddev/tb/grub-0.5.92/stage2/disk_io.c line 93-94:

```c
  if (byte_len <= 0)
    return 1;
```

So for this case, the function should just `return 1` without setting `errnum`.
Our current function would return 0 in case of `byte_len == 0` instead.
Our current function would break in case of `byte_len < 0` instead.

Reply:
Improved.